starknet/docs/its.md

```diff
@@ -0,0 +1,574 @@
+  -h, --help                         display help for command
+```
+
+**Note:** Unlike other deployment scripts, ITS deployment does not support offline mode or gas estimation as it uses direct contract deployment.
+
+**Example:**
+```bash
+npx ts-node starknet/its/deploy-its.ts \
+  --env testnet \
+  --gateway 0x... \
+  --gasService 0x... \
+  --chainName starknet \
+  --itsHubChainName axelar \
+  --itsHubContractAddress 0x... \
+  --operator 0x... \
+  --owner 0x... \
+  --salt 0x123 \
+  --privateKey 0x... \
+  --accountAddress 0x...
+```
+
+### Deploy InterchainTokenFactory
+
+Deploy the InterchainTokenFactory contract on Starknet. This contract is responsible for creating and managing interchain tokens.
+
+```
+Usage: deploy-itf [options]
+
+Deploy InterchainTokenFactory contract on Starknet
+
+Options:
+  --interchainTokenService <address> InterchainTokenService contract address
+  --owner <address>                  Owner address (defaults to account address)
+  --salt <salt>                      Salt for deployment (defaults to 0)
+                                     (default: "0")
+  -e, --env <env>                    environment (choices: "devnet-amplifier",
+                                     "mainnet", "stagenet", "testnet", default:
+                                     "testnet", env: ENV)
+  -y, --yes                          skip deployment prompt confirmation (env:
+                                     YES)
+  -p, --privateKey < privateKey >    private key for Starknet account(testnet
+                                     only, not required for offline tx
+                                     generation) (env: STARKNET_PRIVATE_KEY)
+  --accountAddress <accountAddress>  Starknet account address (env:
+                                     STARKNET_ACCOUNT_ADDRESS)
+  -h, --help                         display help for command
+```
+
+**Note:** Like ITS deployment, ITF deployment does not support offline mode or gas estimation as it uses direct contract deployment.
+
+**Example:**
+```bash
+npx ts-node starknet/its/deploy-itf.ts \
+  --env testnet \
+  --interchainTokenService 0x... \
+  --owner 0x... \
+  --salt 0x123 \
+  --privateKey 0x... \
+  --accountAddress 0x...
+```
+
+## Token Operations
+
+### Deploy Interchain Token
+
+Deploy a new interchain token locally via the InterchainTokenFactory.
+
+```
+Usage: deploy-token [options]
+
+Deploy a new interchain token on Starknet via InterchainTokenFactory
+
+Options:
+  --salt <salt>                      Salt for token deployment
+  --name <name>                      Token name
+  --symbol <symbol>                  Token symbol
+  --decimals <decimals>              Token decimals
+  --initialSupply <amount>           Initial supply to mint (defaults to 0)
+  --minter <address>                 Minter address (if initial supply > 0,
+                                     mintership will be transferred to this
+                                     address)
```

Comment:
```suggestion
  --minter <address>                 Minter address (mintership will be
                                     transferred to this address)
```

I believe whenever a minter is passed, the mintership is transferred. It reverts in case the initial supply is <= 0 and no minter is passed.